### PR TITLE
Combine test attributes

### PR DESCRIPTION
**Pull request by philipktlin**
Talk with @myagley; agreed to combine E2E to integration attibute.  Replace all E2E attribute usage by Integration attribute; and remove E2E attribute.

### CODE REVIEW

**Comment by philipktlin** [comment]
Will need to update "Azure-IoT-Edge-Core CI" vsts build test filter to use "Integration" instead of "E2E"

File: doc/build-test.md [hunk 1]
```diff
@@ -27,9 +27,10 @@ Test
...
-| Stress      | Specialized end-to-end tests for scenarios under heavy load.                        |                                                                                               | N/A               |
+| Attribute   | Goal                                                                                            | Notes                                                                                 | Runs in build(s)  |
+|-------------|-------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------|-------------------|
+| Unit        | Fast, reliable, repeatable tests against low-level components of the codebase.                  | Should not depend on external resources (e.g., network, disk) or configuration.       | Checkin           |
+| Integration | Tests that verify interaction with dependencies _in isolation_and core end-to-end scenarios.    | Require external configuration.                                                       | CI                |
```
Comment (myagley) [inline doc/build-test.md]:
```suggestion
| Integration | Tests that verify interaction with dependencies _in isolation_ and core end-to-end scenarios.    | Require external configuration.                                                       | CI                |
```

File: edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/DeviceScopeIdentitiesCacheTest.cs [hunk 2]
```diff
@@ -240,8 +240,8 @@ public async Task RefreshCacheWithRefreshRequestTest()
             deviceScopeIdentitiesCache.InitiateCacheRefresh();
             deviceScopeIdentitiesCache.InitiateCacheRefresh();
 
-            // Wait for the 2 refresh cycles to complete, this time because of the refresh request
-            await Task.Delay(TimeSpan.FromSeconds(2));
+            // Wait for the 5 refresh cycles to complete, this time because of the refresh request
```
Comment (varunpuranik) [inline edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/DeviceScopeIdentitiesCacheTest.cs]: This should be 2, not 5

Comment (varunpuranik) [reply edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/DeviceScopeIdentitiesCacheTest.cs]: It is not the the time, but the number of times we try..

Comment (philipktlin) [reply edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/DeviceScopeIdentitiesCacheTest.cs]: reverted

**Pull request merged**